src/controllers/destination.ts

```diff
@@ -0,0 +1,169 @@
+import { Context } from 'koa';
+import MiscService from '../services/misc';
+import PreTransformationDestinationService from '../services/destination/preTransformation';
+import PostTransformationDestinationService from '../services/destination/postTransformation';
+import {
+  ProcessorTransformationRequest,
+  RouterTransformationRequestData,
+  RouterTransformationRequest,
+  ProcessorTransformationResponse,
+} from '../types/index';
+import ServiceSelector from '../helpers/serviceSelector';
+import ControllerUtility from './util';
+import stats from '../util/stats';
+import logger from '../logger';
+import { getIntegrationVersion } from '../util/utils';
+import tags from '../v0/util/tags';
+
+export default class DestinationController {
+  public static async destinationTransformAtProcessor(ctx: Context) {
+    const startTime = new Date();
+    logger.debug(
+      'Native(Process-Transform):: Requst to transformer::',
+      JSON.stringify(ctx.request.body),
+    );
+    let resplist: ProcessorTransformationResponse[];
+    let requestMetadata = MiscService.getRequestMetadata(ctx);
+    let events = ctx.request.body as ProcessorTransformationRequest[];
+    const metaTags = MiscService.getMetaTags(events[0].metadata);
+    const { version, destination }: { version: string; destination: string } = ctx.params;
+    const integrationService = ServiceSelector.getDestinationService(events);
+    try {
+      integrationService.init();
+      events = PreTransformationDestinationService.preProcess(
+        events,
+        ctx,
+      ) as ProcessorTransformationRequest[];
+      resplist = await integrationService.processorRoutine(
+        events,
+        destination,
+        version,
+        requestMetadata,
+      );
+    } catch (error) {
+      resplist = events.map((ev) => {
```

Comment:
Is it possible to create a function for this as it is repeated?

Reply:
Actually the processor and router handling has difference we can create two functions but already we are using `getTags` other aspect is different for proc and router flow for adding metadata infos